src/command-dispatcher.ts

```diff
@@ -0,0 +1,82 @@
+import {  } from "./server";
+
+const {createUsersTable, createUser, createLikedLanguagesTable, 
+    addLikedLanguage, getUsers} = require('./server');
+
+//Enumerator with all the commands available
+enum commandsList {
+    CREATE_USER_TABLE = "newusertable",
+    CREATE_LANGUAGES_TABLE = "newlanguagestable",
+    ADD_USER = "newuser",
+    ADD_LIKED_LANGUAGE = "addlanguage",
+    LIST_USERS = "listusers"
+}
+
+//arguments passed with the command
+const args: string[] = (process.argv.slice(2));
+
+/* Entry point of the program.
+   Processes the commands received and calls
+   server functions
+ */
```

Comment:
Hey @fmnjose , can you modify the code so that after some operations, like `createUser` or `addLikedLanguage` a call is made to get all the entries on the database, and show them?
Besides the code that implements each of those functionalities, what did you have to change in order to guarantee the proper sequence of operations?

Reply:
I'm trying to solve this issue together with https://github.com/fmnjose/lovelystay-test/pull/5#discussion_r840358555 ,
but I'm struggling with the bluebird library.

My idea was to have all the functions from server.ts as promises, and dispatcher.js would call them and terminate the program when the async call is finished ( either resolved or rejected).

This would also allow to chain functions from server.ts. For example, when executing the createUser function, i would chain to it the getUsers function, obtaining a promise chain that will create the user and only list all entries after the user creation (considering it is successful).

I will keep chasing this, but will commit the changes to the other comments in the meantime.

Reply:
Bluebird promises finally clicked!
Will commit changes done to support async functions and db calls chaining.